org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/license/LicenseUtilStatisticsTest.java

```diff
@@ -0,0 +1,161 @@
+        Assert.assertEquals("candidateLicenseFileCount", 0, generalStatistics.getCandidateLicenseFileCount());
+        Assert.assertEquals("totalArchiveCount", 0, generalStatistics.getTotalArchiveCount());
+    }
+
+    /**
+     * Test for {@link LicenseUtil#calculateGeneralStatistics(Collection)}.
+     */
+    @Test
+    public void testCalculateGeneralStatisticsRealList() {
+        final Collection<Archive> archives = new ArrayList<Archive>();
+        for (int i = 1; i <= 3; i++) {
+            Archive archive = new Archive(ArchiveType.JAVA, "archive" + i, "", "");
+            for (int j = 1; j <= i; j++) {
+                archive.addLicenseCandidateFile("candidateFile" + j);
+            }
+            archives.add(archive);
+        }
+        final GeneralStatistics generalStatistics = LicenseUtil.calculateGeneralStatistics(archives);
+        Assert.assertEquals("candidateLicenseFileCount", 6, generalStatistics.getCandidateLicenseFileCount());
+        Assert.assertEquals("totalArchiveCount", 3, generalStatistics.getTotalArchiveCount());
+    }
+
+    /**
+     * Test for {@link LicenseUtil#calculateDetectionStatusStatistics(Collection)}.
+     */
+    @Test
+    public void testCalculateDetectionStatusStatisticsEmptyList() {
+        final Collection<Archive> archives = new ArrayList<Archive>();
+        final IDetectionStatusStatistics detectionStatusStatistics = LicenseUtil
+                .calculateDetectionStatusStatistics(archives);
+        Assert.assertEquals("detectionStatusStatistics (DETECTED)", 0,
+                (int) detectionStatusStatistics.get(DetectionStatus.DETECTED));
+        Assert.assertEquals("detectionStatusStatistics (MANUAL_DETECTED)", 0,
+                (int) detectionStatusStatistics.get(DetectionStatus.MANUAL_DETECTED));
+        Assert.assertEquals("detectionStatusStatistics (MANUAL_NOT_DETECTED)", 0,
+                (int) detectionStatusStatistics.get(DetectionStatus.MANUAL_NOT_DETECTED));
+        Assert.assertEquals("detectionStatusStatistics (MANUAL_SELECTED)", 0,
+                (int) detectionStatusStatistics.get(DetectionStatus.MANUAL_SELECTED));
+        Assert.assertEquals("detectionStatusStatistics (MULTIPLE_DETECTED)", 0,
+                (int) detectionStatusStatistics.get(DetectionStatus.MULTIPLE_DETECTED));
+        Assert.assertEquals("detectionStatusStatistics (NOT_DETECTED)", 0,
+                (int) detectionStatusStatistics.get(DetectionStatus.NOT_DETECTED));
+    }
+
+    /**
+     * Test for {@link LicenseUtil#calculateDetectionStatusStatistics(Collection)}.
+     */
+    @Test
+    public void testCalculateDetectionStatusStatisticsRealList() {
+        final Collection<Archive> archives = new ArrayList<Archive>();
+        for (DetectionStatus detectionStatus : DetectionStatus.values()) {
+            final int ordinal = detectionStatus.ordinal();
+            for (int i = 0; i < ordinal; i++) {
```

Comment:
Use `<=` here to add at least one of archive with status `DETECTED` (whose ordinal is 0):

```suggestion
            for (int i = 0; i <= ordinal; i++) {
```

NOTE: if so, every assertion expected value should be updated by 1.

Reply:
This is intentional to test a case where there is no archive of a certain status. However, added a comment explaining this.